Backpack-SwiftUI/Calendar/Classes/Core/CalendarMonthGrid.swift

```diff
@@ -40,56 +40,74 @@ struct CalendarMonthGrid<
         let firstWeekday = calendar.firstWeekday // Locale-aware first day of the week
         let weekdayOfMonthStart = calendar.component(.weekday, from: monthDate)
         // Calculate the offset based on the first weekday
-        let weekdaysOffset = (weekdayOfMonthStart - firstWeekday + daysInAWeek) % daysInAWeek
-        let daysFromPreviousMonth = weekdaysOffset
-        let emptyDaysLeading = (0..<daysFromPreviousMonth).map { index in
-            EmptyCellInfo(cellIndex: index, month: monthDate)
-        }
-        
-        let numberOfDaysInMonth = calendar.range(of: .day, in: .month, for: monthDate)!.count
-        let totalCellsUsed = numberOfDaysInMonth + daysFromPreviousMonth
-        let remainingCells = daysInAWeek - (totalCellsUsed % daysInAWeek)
-        let emptyDaysTrailing = (0..<remainingCells).map { index in
-            EmptyCellInfo(cellIndex: index, month: monthDate)
-        }
+        let daysFromPreviousMonth = (weekdayOfMonthStart - firstWeekday + daysInAWeek) % daysInAWeek
 
         LazyVGrid(
             columns: Array(repeating: GridItem(spacing: BPKSpacing.none.value), count: daysInAWeek),
             spacing: BPKSpacing.lg.value
         ) {
             // Create cells for the days from the previous month that are shown in the first week of the current month.
-            ForEach(emptyDaysLeading) { emptyDayInfo in
-                VStack(spacing: BPKSpacing.none) {
-                    emptyLeadingDayCell(emptyDayInfo)
-                        .frame(height: dayCellHeight)
-                    Spacer(minLength: BPKSpacing.none)
-                }
-            }
-
+            previousEmptyCells(daysFromPreviousMonth: daysFromPreviousMonth)
+            let numberOfDaysInMonth = calendar.range(of: .day, in: .month, for: monthDate)!.count
             // Create cells for the days in the current month
             currentMonthDayCell(numberOfDaysInMonth: numberOfDaysInMonth)
 
             // Create cells for the days from the next month that are shown in the last week of the current month
             // The total number of cells used is the sum of the number of days in the current month and the number of
             // days from the previous month that are shown
-            
-            if remainingCells < daysInAWeek {
-                ForEach(emptyDaysTrailing) { emptyDayInfo in
-                    VStack(spacing: BPKSpacing.none) {
-                        emptyTrailingDayCell(emptyDayInfo)
-                            .frame(height: dayCellHeight)
-                        Spacer(minLength: BPKSpacing.none)
-                    }
+            let totalCellsUsed = numberOfDaysInMonth + daysFromPreviousMonth
+            let remainingCells = daysInAWeek - (totalCellsUsed % daysInAWeek)
+        
+            remainingEmptyCells(remainingCells: remainingCells)
+        }
+    }
+    
+    @ViewBuilder
+    private func previousEmptyCells(daysFromPreviousMonth: Int) -> some View {
```

Comment:
Extracted this from the above method to make it easier to read, also introducing the usage of a small internal struct DayCellIdentifiable, as the preiovus range only approach weas leading to mis-reusage of some cells, as they had repeating ids (which was a bug).
Now we can set specific IDs to each cell and make sure they do not repeat across weeks/months in the calendar, fixing the bug where sometimes the cells were not rendering correctly